MRG: Fix imsave

larsoner opened the pull request:
Closes #209.

@mwaskom I used your approach but left the figure-generation private for now. Feel free to make a follow-up PR to make it public with proper documentation, etc.

mwaskom commented:
Haven't tested but LGTM

mwaskom commented on surfer/viz.py:
```diff
@@ -2652,7 +2652,7 @@ def save_movie(self, fname, time_dilation=4., tmin=None, tmax=None,
                              "(%r)" % (tmax, self._times[-1]))
         n_frames = floor((tmax - tmin) * time_dilation * framerate)
-        times = np.arange(n_frames)
+        times = np.arange(n_frames).astype(float)
```
Probably not a big performance hit here, but you could pass the dtype to arange to avoid making a copy.

larsoner commented:
Okay CIs are happy (except pending OSX but that usually takes forever). I added a 3.6 CI, fixed related errors, and fixed a couple of bare `except:`s we had. Ready for review/merge from my end.

larsoner commented:
(actually not sure about AppVeyor yet, either, but it passed last time I let it finish)

christianbrodbeck commented:
Is there an advantage over using Matplotlib's `imsave` (https://github.com/matplotlib/matplotlib/blob/5f63dd42ec6deec5c85ab975d7d879c7c81cb161/lib/matplotlib/image.py#L1305), which seems to be more efficient for PNG but do the same otherwise?

larsoner commented:
I assumed it was just an efficiency thing so I didn't include it to reduce complexity, but I don't know for sure

christianbrodbeck commented:
Oh, I didn't mean to include the special case, but whether it wouldn't be easier to just use `matplotlib.image.imsave`

larsoner commented:
This moves toward a use case @mwaskom wanted, see his comments toward the end of #209

christianbrodbeck commented:
Aah, makes sense

> On Nov 27, 2017, at 9:21 PM, Eric Larson <notifications@github.com> wrote:
> 
> This moves toward a use case @mwaskom <https://github.com/mwaskom> wanted, see his comments toward the end of #209 <https://github.com/nipy/PySurfer/issues/209>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub <https://github.com/nipy/PySurfer/pull/221#issuecomment-347392650>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAI5a5pmweM1CpgZiXtKpgzimeP7t1k6ks5s624igaJpZM4Qr8Mk>.
>

larsoner commented:
Anybody else want to look (or @mwaskom want to look again now that everything passes)? If not I'll merge at the end of the week
